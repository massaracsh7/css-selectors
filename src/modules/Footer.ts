import { PartsPage } from './types';
import Block from './Block';
import Link from './Link';
import Element from './Element';

export default class Footer implements PartsPage {
  view = (): HTMLElement => {
    const logoImg = document.createElement('img');
    logoImg.src = 'https://rs.school/images/rs_school_js.svg';
    const linkGit = new Link('footer__git', 'github', 'https://github.com/massaracsh7');
    const copy = new Element('div', 'footer__year', '2023');
    const linkRss = new Link('footer__rss', '', 'https://rs.school/js/').create();
    linkRss.append(logoImg);
    const footer = new Block('div', 'footer', linkGit.create(), copy.create(), linkRss);
    return footer.create();
  };
}
