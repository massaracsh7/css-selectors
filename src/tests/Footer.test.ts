/** @type {import('<rootDir>/node_modules/ts-jest/dist/types').InitialOptionsTsJest} */

import Footer from '../modules/Footer';

describe('Footer class', () => {
  let footer: HTMLElement;

  beforeEach(() => {
    footer = new Footer().view();
  });

  it('should return a defined footer element', () => {
    expect(footer).toBeDefined();
    expect(footer).toBeInstanceOf(HTMLElement);
  });

  it('should match the snapshot', () => {
    expect(footer).toMatchSnapshot();
  });

  it('should have the correct tag name', () => {
    expect(footer.tagName).toBe('FOOTER');
  });

  it('should contain the expected classes', () => {
    expect(footer.classList.contains('footer')).toBe(true);
  });

});
