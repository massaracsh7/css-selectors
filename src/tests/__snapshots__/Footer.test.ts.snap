// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer class should match the snapshot 1`] = `
<footer
  class="footer"
>
  <a
    class="footer__git"
    href="https://github.com/massaracsh7"
  >
    github
  </a>
  <div
    class="footer__year"
  >
    2023
  </div>
  <a
    class="footer__rss"
    href="https://rs.school/js/"
  >
    <img
      src="https://rs.school/images/rs_school_js.svg"
    />
  </a>
</footer>
`;
